### ISSUE TITLE
Problems with making predictions using the already trained model

### ISSUE THREAD

**Issue by ognjenkundacina** [BNN-UPC/ignnition#87]
Hi ignnition team!

Briefly the problem is that model.predict() function doesn't return good results when it is called without model.train_and_validate() function.

I've trained a GNN model successfully (training and validation losses were converging to zero) and predictions from the predict() methods were aligned with the labels in the prediction set. I've used the following code in the main() method: 
    model = ignnition.create_model(model_dir='./')
    model.computational_graph()
    model.train_and_validate()
    predictions = model.predict(num_predictions = 1)

Part of the train_options.yaml file:
train_dataset: ./data/train
validation_dataset: ./data/test
predict_dataset: ./data/test
load_model_path: ./weights.1000-0.00R.hdf5
additional_functions_file: ./main.py
output_path: ./

I copied trained trained model parameters "weights.1000-0.00R.hdf5" from the CheckPoint into the root directory and called the model.predict() in the following way:
    model = ignnition.create_model(model_dir='./')
    predictions = model.predict(num_predictions = 1)

and in this way also:
    model = ignnition.create_model(model_dir='./')
    model.computational_graph()
    predictions = model.predict(num_predictions = 1)

but the predictions were not fitting the labels in the predict set well. I also tried calling the model.train_and_validate() function with epochs and epoch_size set to 0, but it gave the same results. 

I would also note that there is no "weights.1000-0.00R.hdf5" in some location other than root directory, so the correct trained parameters should be loaded:

Console logs:
←[1m
Processing the described model...
←[0m
←[1mCreating the GNN model...
←[0m
Restoring from ./weights.1000-0.00.hdf5
←[1mStarting to make the predictions...

You can find the code attached in code.zip, as well as predicted vs label plots for both working and non-working examples.

Kind regards!

Ognjen

[code.zip](https://github.com/BNN-UPC/ignnition/files/7333522/code.zip)
![working_example](https://user-images.githubusercontent.com/37520582/137038936-f1557695-9f45-409e-a4c8-394784cd5ca2.png)
![non_working_example](https://user-images.githubusercontent.com/37520582/137038945-f735fed8-1202-4916-8a75-8a5acebfeb48.png)

**Comment by MiquelFerriol** [comment]
Dear @ognjenkundacina,
First of all, thank you very much for the time you are spending working with IGNNITION and for asking/reporting questions/problems you may found. This really helps us to constantly improve our framework.

We found a problem in the implementation where the weights weren't correctly saved when having attention as an aggregation function. This caused the model to load all the weights except the ones of the attention causing the discrepancy in the results.

This problem has been fixed in #90. Please, consider updating IGNNITION to the latest version (at this time 1.3.2).

Note that the HDF5 format has been deprecated, and now the framework saves the weights in a more standard way. In order to load a pre-trained model you now need to define the load_model_path in the train_options.yaml without the ckpt extension doing:
`load_model_path: ./CheckPoint/experiment_2021_10_13_20_28_19/ckpt/weights.10-0.001`
and **NOT**:
`load_model_path: ./CheckPoint/experiment_2021_10_13_20_28_19/ckpt/weights.10-0.001.index`

Thanks again for your time and we are sorry for the problems this inconvenience has possibly caused.

If you have any doubt/problem, please do not hesitate to contact us again.

Bests,
Miquel

**Comment by ognjenkundacina** [comment]
Thank you all for your detailed and prompt responses! I find this framework very helpful, I'll continue using it for sure.

I've upgraded ignnition to the newest version and the issue doesn't reproduce anymore, so it can be closed.
